dedupe/datamodel.py

```diff
@@ -214,16 +216,12 @@ def missing(variables: list[Variable]) -> list[MissingDataType]:
     return missing_variables
 
 
-def interactions(
-    definitions: Iterable[VariableDefinition], primary_variables: list[FieldVariable]
-) -> list[InteractionType]:
-    var_d = {var.name: var for var in primary_variables}
-
+def _expanded_interactions(variables: list[Variable]) -> list[InteractionType]:
+    field_vars = {var.name: var for var in variables if isinstance(var, FieldVariable)}
```

Comment:
Again, this should feels like it needs a more official API, or should be responsibility of the InteractionType

Reply:
i can think of a few designs here, but this code is actually doing two things for us, both of which are kind of important.

1. isolating the interaction features for us, so we can put them in the right place in the sequence of features. 
2. actually resolving the feature indices. 

seems like we kind of need to do both.